src/errors.ts

```diff
@@ -9,3 +9,43 @@ export class RejectRequestError extends Error {
         super('Request was rejected by the wallet.');
     }
 }
+
+export class UnauthorizedError extends Error {
+    public code?: string;
+    constructor(code?: string) {
+        super(code || 'Unauthorized');
+        this.code = code;
+    }
+}
+
+const UNAUTH_CODES = new Set(['UNAUTHENTICATED', 'UNAUTHORIZED', 'SESSION_EXPIRED', 'LOGGED_OUT']);
+
+export function extractErrorCode(message: any): string | null {
+    if (message?.type === 'unauthorized' && typeof message?.code === 'string') {
+        return message.code;
+    }
+
+    const candidates = [
+        message?.error?.code,
```

Comment:
can we standarize on of our error to a single type? like just `message.error.code` ? Currently our API is still small and easy to standarize all ?